sdk/attestation/azure-security-attestation/src/private/attestation_client_private.hpp

```diff
@@ -231,7 +231,7 @@ namespace Azure { namespace Security { namespace Attestation { namespace _detail
      *
      * @param jwt - the JSON Web Token/JSON Web Signature to be parsed.
      */
-    AttestationTokenInternal(std::string const& jwt, Azure::Nullable<T> preferredBody = {})
+    AttestationTokenInternal(std::string const& jwt, T const* const preferredBody = nullptr)
```

Comment:
the preferredBody is not mentioned in the doxygent comments. It's not required here for non-public.
But I would still like to know what it is used for.

Reply:
Comment added. Basically this allows a creator of an AttestationTokenInternal to create an AttestationToken whose body doesn't match the object contained in the JWT.

The primary use for this is in the GetPolicy API - the JWT contains a StoredAttestationPolicy object, which has a single value AttestationPolicy, which is a Base64Url encoded version of the policy document. That complexity is necessary for the wire protocol but is unnecessary for SDK customers, so the GetPolicy API returns an AttestationToken<std::string>.

To make that happen, `preferredBody` is set to the std::string value to be returned to the customer.